src/api/store/userprofile.py

```diff
@@ -67,20 +69,64 @@ def add_household(self, context: Context, args) -> (dict, MassEnergizeAPIError):
       user = get_user_or_die(context, args)
       name = args.pop('name', None)
       unit_type=args.pop('unit_type', None)
+      # Location is string address of the unit, deliminted as follows:
+      # 'street' + ", " + city + ", " + state + ", " + 'zipcode'
       location=args.pop('location', None)
-      communityId = args.pop('community_id', None) or args.pop('community', None) 
-
-      new_unit = RealEstateUnit.objects.create(name=name, unit_type=unit_type,location=location)
-      new_unit.save()
-
-      user.real_estate_units.add(new_unit)
-      if communityId:
-        community = Community.objects.get(id=communityId)
-        new_unit.community = community
-
-      new_unit.save()
+      address = args.pop('address', None)
+      if address:
+        address = json.loads(address)
+        street = address.get('street', '')
+        unit_number = address.get('unit_number', '')
+        zipcode = address.get('zipcode', '')
+        city = address.get('city', '')
+        county = address.get('county', '')
+        state = address.get('state', '')
+      else:
+        # get address from location string
+        loc_parts = location.capitalize().replace(" ","").split(',')
```

Comment:
i really really think you should have a helper function defined somewhere that does this work.  I could see other functions needed this.

Reply:
Good point, and I think that showed a potential bug since it wasn't identical in the three places.  Now I've made a helper function split_location_string